clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -514,22 +514,32 @@ private void maybeAutoCommitOffsetsAsync(long now) {
                 this.nextAutoCommitDeadline = now + retryBackoffMs;
             } else if (now >= nextAutoCommitDeadline) {
                 this.nextAutoCommitDeadline = now + autoCommitIntervalMs;
-                commitOffsetsAsync(subscriptions.allConsumed(), new OffsetCommitCallback() {
-                    @Override
-                    public void onComplete(Map<TopicPartition, OffsetAndMetadata> offsets, Exception exception) {
-                        if (exception != null) {
-                            log.warn("Auto offset commit failed for group {}: {}", groupId, exception.getMessage());
-                            if (exception instanceof RetriableException)
-                                nextAutoCommitDeadline = Math.min(time.milliseconds() + retryBackoffMs, nextAutoCommitDeadline);
-                        } else {
-                            log.debug("Completed autocommit of offsets {} for group {}", offsets, groupId);
-                        }
-                    }
-                });
+                maybeAutoCommitOffsetsNow(true);
             }
         }
     }
 
+    /**
+     * @param bypassChecks: whether the auto-commit and coordinator status can be
+     * bypassed or have to be double-checked
+     */
+    public void maybeAutoCommitOffsetsNow(boolean bypassChecks) {
```

Comment:
@hachikuji I added a boolean parameter so we don't re-evaluate the checks already performed in the existing private method. Please advise if you think we should go about this in a different way. Thanks.